src/textEditor.ts

```diff
@@ -140,6 +141,32 @@ export class TextEditor {
     return vscode.window.activeTextEditor.document.getText(selection);
   }
 
+  /**
+   *  Retrieves the current word at position.
+   *  If current position is whitespace, selects the right-closest word
+   */
+  static getWord(position: Position) : string {
+    const whitespaceRegex = new RegExp("^ *$");
+    let start = position;
+    let end = position.getRight();
+
+    const char = TextEditor.getText(new vscode.Range(start, end));
+    if (whitespaceRegex.test(char)) {
+      start = position.getWordRight();
+    } else {
+      start = position.getWordLeft(true);
+    }
+    end = start.getCurrentWordEnd(true).getRight();
+
+    const word = TextEditor.getText(new vscode.Range(start, end));
+
+    if (whitespaceRegex.test(word)) {
+      throw VimError.fromCode(ErrorCode.E348);
```

Comment:
I'm beginning to think we should move away from `throw` and `try/catch`. The more I use them the more I feel like they're just a goto statement to some arbitrary point in the codebase. 

I'd much rather prefer making the return type something like `string | undefined` to indicate failure and then handling that at the call site.